python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets.py

```diff
@@ -673,3 +679,158 @@ def foo():
+        [
+            AssetsDefinition.from_graph(my_graph, can_subset=True),
+        ],
+        [],
+    )
+
+    with instance_for_test() as instance:
+        asset_job.execute_in_process(instance=instance, asset_selection=[AssetKey("one")])
+        materialization_planned = list(
+            instance.get_event_records(
+                EventRecordsFilter(DagsterEventType.ASSET_MATERIALIZATION_PLANNED)
+            )
+        )
+        assert len(materialization_planned) == 1
+        step_keys = get_step_keys_from_run(instance)
+        assert set(step_keys) == set(["my_graph.foo", "my_graph.bar_1"])
+
+
+def test_graph_backed_asset_partial_output_selection():
+    @op(out={"a": Out(), "b": Out()})
+    def foo():
+        return 1, 2
+
+    @graph(out={"one": GraphOut(), "two": GraphOut()})
+    def graph_asset():
+        one, two = foo()
+        return one, two
+
+    asset_job = define_asset_job("yay").resolve(
+        [
+            AssetsDefinition.from_graph(graph_asset, can_subset=True),
+        ],
+        [],
+    )
+
+    with instance_for_test() as instance:
+        asset_job.execute_in_process(instance=instance, asset_selection=[AssetKey("one")])
+        materialization_planned = list(
+            instance.get_event_records(
+                EventRecordsFilter(DagsterEventType.ASSET_MATERIALIZATION_PLANNED)
+            )
+        )
+        assert len(materialization_planned) == 1
+        step_keys = get_step_keys_from_run(instance)
+        assert set(step_keys) == set(["graph_asset.foo"])
+
+
+def test_input_subsetting_graph_backed_asset():
+    @asset
+    def upstream_1():
+        return 1
+
+    @asset
+    def upstream_2():
+        return 1
+
+    @op
+    def bar(foo):
+        return foo
+
+    @op
+    def baz(up_1, up_2):
+        return up_1 + up_2
+
+    @graph(out={"one": GraphOut(), "two": GraphOut(), "three": GraphOut()})
+    def my_graph(upstream_1, upstream_2):
+        return (
+            bar.alias("bar_1")(upstream_1),
+            bar.alias("bar_2")(upstream_2),
+            baz(upstream_1, upstream_2),
+        )
+
+    asset_job = define_asset_job("yay").resolve(
+        [
+            upstream_1,
+            upstream_2,
+            AssetsDefinition.from_graph(my_graph, can_subset=True),
+        ],
+        [],
+    )
+
+    # test first "bar" alias
+    with instance_for_test() as instance:
+        asset_job.execute_in_process(
+            instance=instance,
+            asset_selection=[AssetKey("one"), AssetKey("upstream_1"), AssetKey("upstream_2")],
+        )
+        materialization_planned = list(
+            instance.get_event_records(
+                EventRecordsFilter(DagsterEventType.ASSET_MATERIALIZATION_PLANNED)
+            )
+        )
+        assert len(materialization_planned) == 3
+        step_keys = get_step_keys_from_run(instance)
+        assert set(step_keys) == set(["my_graph.bar_1", "upstream_1", "upstream_2"])
```

Comment:
For this test, I'm curious about what would happen if the asset selection was just "one" and "upstream_1". I think that would better match the name of "test_input_subsetting", as in this case only one of the input to the graph would be used.

Fine with this test sticking around, just would also like to see that other behavior under test.